### ISSUE TITLE
Missing label for empty option in Country list

### ISSUE THREAD

**Issue by metamoni** [Shopify/dawn#2135]
**Describe the current behavior**
When a user goes to update their address and navigates to the Country dropdown, the `---` option doesn't have an accessible label. Without this, the screen reader doesn't know what to announce.

**Describe the expected behavior**
Screen readers should announce something like "No country selected" or "Empty".

**Version information (Dawn, browsers and operating systems)**
- Dawn Version: 7.0.1
- Chrome Version 107.0.5304.110
- macOS Version 13.0.1

**Possible solution**
One of these solutions should work:
- in the specified `option`, add a visually-hidden label: `<option ...>--- <span class="visually-hidden">No country selected</span></option>`
- hide the `option` from screen readers: `<option aria-hidden="true">---</option>`
- remove the option altogether, as [selecting it triggers a validation error](https://github.com/Shopify/dawn/pull/2118) anyway (should check with design before doing this)

**Additional context/screenshots**

https://user-images.githubusercontent.com/22390758/202646035-8bd89a57-a8cc-4525-934f-fae381f61e50.mp4
